cpp/oneapi/dal/test/engine/dataframe_common.cpp

```diff
@@ -219,6 +219,36 @@ class dataframe_builder_action_allocate : public dataframe_builder_action {
     std::int64_t column_count_;
 };
 
+class dataframe_builder_action_read_external_dataset : public dataframe_builder_action {
+public:
+    explicit dataframe_builder_action_read_external_dataset(std::string dataset)
+            : dataset_(dataset) {}
```

Comment:
I suggest to introduce function `normalize_path` that removes trailing `/` in the beginning or end of file path.